packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1372,18 +1372,33 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             }
 
             if (!this.runtimeOptions.disableGC) {
-                // Get the container's GC data and run GC on the reference graph in it.
-                const gcData = await this.dataStores.getGCData();
-                const { referencedNodeIds } = runGarbageCollection(gcData.gcNodes, [ "/" ], this.logger);
-
-                // Update our summarizer node's used routes. Updating used routes in summarizer node before summarizing
-                // is required and asserted by the the summarizer node. We are the root and are always referenced, so
-                // the used routes is only self-route (empty string).
-                this.summarizerNode.updateUsedRoutes([""]);
-
-                // Remove this node's route ("/") and notify data stores of routes that are used in it.
-                const usedRoutes = referencedNodeIds.filter((id: string) => { return id !== "/"; });
-                this.dataStores.updateUsedRoutes(usedRoutes);
+                const perfEvent = PerformanceEvent.start(summaryLogger, {
+                    eventName: "GarbageCollection",
+                });
+
+                const gcStats: { totalGCNodes?: number; deletedGCNodes?: number } = {};
+                try {
+                    // Get the container's GC data and run GC on the reference graph in it.
+                    const gcData = await this.dataStores.getGCData();
+                    const { referencedNodeIds } = runGarbageCollection(gcData.gcNodes, [ "/" ], this.logger);
+
+                    // Update stats to be reported in the peformance event.
+                    gcStats.totalGCNodes = Object.keys(gcData.gcNodes).length;
```

Comment:
Not sure if it will be significant, but Object.keys might not be cheap if the number is large.

Reply:
Updated to use the result of runGarbageCollection and not use Object.keys.